test/equalStrengthLeftTests.html

```diff
@@ -0,0 +1,52 @@
+<!-- Max Tarlov 8/12/19 testing abstract match constraints (Kalivoda) -->
+<html>
+	<title>Abstract Match Test Left</title>
+	<head>
+		<link rel="stylesheet" type="text/css" href="../spot.css">
+		<script src="../build/spot.js"></script>
+		<script src="../build/constraints.js"></script>
+		<script src="../candidategenerator.js"></script>
+	</head>
+	<body>
+		<pre id="results-container"></pre>
+
+		<script>
+			var sTree ={
```

Comment:
Is this the only stree you tested with? I just tested it with an stree with more stacked up edges: [[[[a]b]c]d] vs. {((a)b) c d} and that seemed to work, but it would be good to test with some additional, methodically constructed stree/ptree combinations as well.

Definitely check whether it's working properly with different categories, too, because my preliminary test with {{(a) b} c d} + [[[[a]b]c]d] did not give me any errors.

Reply:
So I tested with a lot of different trees, but mostly in console so they aren't represented here. It looks like what happened was that you used a prosodic category when the function required a syntactic category, which doesn't break the code, it just means that the base functions are counting the number of iotas syntactic terminals are falling on and the number of CPs prosodic categories are falling on.
Since this is a mistake that I also made in testing, I added a little code bloc to convert prosodic categories to syntactic categories so that either can be used in equalStrengthLeft and equalStrengthRight.